src/main/java/net/mcreator/element/types/interfaces/IItemWithModel.java

```diff
@@ -21,12 +21,18 @@
 
 import net.mcreator.workspace.resources.Model;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.Map;
 
-@SuppressWarnings("unused") public interface IItemWithModel {
+@SuppressWarnings("unused") public interface IItemWithModel extends IResourcesDependent {

```

Comment:
Interfaces like this one should not extend IResourcesDependent/IOtherModElementsDependents/imilar interfaces added by this PR. Most MEs use more than one of those and thus methods of IResourcesDependent/IOtherModElementsDependent will have to be overridden again by said ME.

It also makes code harder to read. Only direct elements (Armor/Biome/...) directly should override IOtherModElementsDependent and provide direct list of all elements there without calling super methods, so one always has list of all used references directly at hand in the code of said element.

Reply:
✔️ Fixed